packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -513,11 +543,45 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
   }
 
   private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+    if (props.activeTracing === undefined &&
+      !props.environmentVariables &&
+      !props.memory &&
+      !props.timeout) {
       return undefined;
     }
+    const activeTracingNotSupportedRuntime = [
```

Comment:
The error messages are taken directly from the [documentation](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-activetracing).
However, since syn-nodejs-2.0 and earlier versions are no longer available, I am only setting it for the currently available runtimes that do not support active tracing (see also [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_Library.html#CloudWatch_Synthetics_Canaries_runtime_support)).

I was unsure about including this validation, so if you have any thoughts or feedback, please let me know.

Reply:
What about checking `props.runtime.family !== RuntimeFamily.NODEJS` like in the other [PR](https://github.com/aws/aws-cdk/pull/30197)?

Reply:
Thank you. I made the correction since checking the runtime is better. 
(Even though I did it myself in another PR...lol)